### PR TITLE
feat: Support `DataFrame` and `Series` init from torch `Tensor` objects

### LINKED ISSUE #17453: Can we add `from_torch`?
### Description

We have `to_torch` for `DataFrame` and `Series`. Can we add `from_torch` to make it more consistent?

### PR DESCRIPTION
**Pull request by alexander-beedie**
Closes #17453.
Closes #19120 (this PR supersedes it).

* Adds support for efficient Polars init from torch Tensors; takes advantage of zero-copy numpy intermediary conversion, so is cheap by default. Tensor detection is lazy.

## Example

```python
import polars as pl
import torch

t = torch.tensor([
  [1234.5, 200.0, 3000.5],
  [8000.0, 500.5, 6000.0],
])

df = pl.from_torch(t, schema=["colx", "coly"])
# shape: (3, 2)
# ┌────────┬────────┐
# │ colx   ┆ coly   │
# │ ---    ┆ ---    │
# │ f32    ┆ f32    │
# ╞════════╪════════╡
# │ 1234.5 ┆ 8000.0 │
# │ 200.0  ┆ 500.5  │
# │ 3000.5 ┆ 6000.0 │
# └────────┴────────┘
```
Can also pass directly to `pl.DataFrame` init.

### CODE REVIEW

**Comment by alexander-beedie** [comment]
(Rebased)

File: py-polars/polars/functions/lit.py [hunk 1]
```diff
@@ -173,25 +182,21 @@ def lit(
     if dtype:
         return wrap_expr(plr.lit(value, allow_object, is_scalar=True)).cast(dtype)
 
-    try:
-        # numpy literals like np.float32(0) have item/dtype
-        item = value.item()
-
-        # numpy item() is py-native datetime/timedelta when units < 'ns'
-        if isinstance(item, (datetime, timedelta)):
+    if isinstance(value, np.generic):
```
Comment (ritchie46) [inline py-polars/polars/functions/lit.py:185]: This made numpy a required package.

Comment (alexander-beedie) [reply py-polars/polars/functions/lit.py:185]: Fixed: https://github.com/pola-rs/polars/pull/22486 ✌️
